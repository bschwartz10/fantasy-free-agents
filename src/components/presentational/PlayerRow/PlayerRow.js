import React from 'react';
import Proptypes from 'prop-types';

const PlayerRow = ({ player }) => {
  return (
    <tr className='player-row'>
      <td className="name">{player.name}</td>
      <td className="position">{player.position}</td>
      <td className="team">{player.team}</td>
      <td className="height">{player.height}</td>
      <td className="weight">{player.weight}</td>
      <td className="adp">{player.adp}</td>
    </tr>
  )
}

PlayerRow.propTypes = {
  player: Proptypes.object,
}
export default PlayerRow;
