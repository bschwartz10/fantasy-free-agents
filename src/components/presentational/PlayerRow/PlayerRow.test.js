import React from 'react';
import { shallow } from 'enzyme';
import PlayerRow from './PlayerRow';

describe('PlayerRow component', () => {
  let wrapper;

  beforeEach(() => {
    const player = {
      adp: "400",
      birthdate: "656226000",
      cbs_id: "1977109",
      college: "UCLA",
      draft_team: "FA",
      draft_year: "2012",
      espn_id: "15351",
      height: "72",
      id: "11524",
      jersey: "48",
      kffl_id: "26333",
      name: "Derrick Coleman",
      nfl_id: "derrickcoleman/2534871",
      position: "RB",
      rotoworld_id: "8019",
      sportsdata_id: "e0a63251-5428-43a1-88c1-c000215ac5ce",
      stats_global_id: "461705",
      stats_id: "26299",
      team: "ARI",
      weight: "233",
    }
    wrapper = shallow(<PlayerRow player={player}/>)
  })

  it('should exist', () => {
    expect(wrapper).toBeDefined()
  })

  it('render player with correct attributes', () => {
    const playerRow = wrapper.find('.player-row')

    expect(playerRow.find('.name').text()).toEqual('Derrick Coleman')
    expect(playerRow.find('.position').text()).toEqual('RB')
    expect(playerRow.find('.team').text()).toEqual('ARI')
    expect(playerRow.find('.height').text()).toEqual('72')
    expect(playerRow.find('.weight').text()).toEqual('233')
    expect(playerRow.find('.adp').text()).toEqual('400')
  })

  it('should match snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

})
