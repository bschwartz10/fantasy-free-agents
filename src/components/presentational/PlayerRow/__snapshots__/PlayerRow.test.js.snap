// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerRow component should match snapshot with all data passed in correctly 1`] = `
<tr
  className="player-row"
>
  <td
    className="name"
  >
    Derrick Coleman
  </td>
  <td
    className="position"
  >
    RB
  </td>
  <td
    className="team"
  >
    ARI
  </td>
  <td
    className="height"
  >
    72
  </td>
  <td
    className="weight"
  >
    233
  </td>
  <td
    className="adp"
  >
    400
  </td>
</tr>
`;
